packages/overlay/imperative-api.md

```diff
@@ -0,0 +1,95 @@
+
+```js
+(async () => {
+    const content = document.querySelector('#content');
+    const options = {
+        offset: 0,
+        placement: 'bottom',
+        trigger: document.querySelector('#trigger'),
+        type: 'auto',
+    };
+    const overlay = await Overlay.open(content, options);
+    document.body.append(overlay);
+})();
+```
+
+Keep in mind that a changing DOM tree can update the relationship between existing content and the CSS that you have applied to it when appending the `<sp-overlay>` element to your existing DOM.
+
+## OverlayOptions
+
+When leveraging `Overlay.open()`, you can provide an optional second argument of `OverlayOptions`, with the following type:
+
+```ts
+type OverlayOptions = {
+    delayed?: boolean;
+    notImmediatelyClosable?: boolean;
+    offset?: number | [number, number];
+    placement?: Placement;
+    receivesFocus?: 'auto' | 'true' | 'false';
+    trigger?: HTMLElement | VirtualTrigger;
+    type?: 'modal' | 'page' | 'hint' | 'auto' | 'manual';
+};
+```
+
+### delayed
+
+An Overlay that is `delayed` will wait until a warm-up period has completed before opening. Once the warmup period has completed, all subsequent Overlays will open immediately. When no Overlays are opened, a cooldown period will begin. Once the cooldown has completed, the next Overlay to be opened will be subject to the warm-up period if provided that option.
+
+### notImmediatelyCloseable
+
+When an Overlay is `notImmediatelyCloseable` that means that the first interaction that would lead to the closure of the Overlay in question will be ignored. This is useful when working with non-"click" mouse interactions, like `contextmenu`, where the trigger event (e.g. `contextmenu`) occurs _before_ an event that would close said overlay (e.g. `pointerup`).
+
+### offset
+
+The `offset` property accepts either a single number, to define the offset of the Overlay along the main axis from the trigger, or 2-tuple, to define the offset along the main axis and the cross axis. This option has no effect when there is no trigger element.
+
+### placement
+
+A `placement` of `"auto" | "auto-start" | "auto-end" | "top" | "bottom" | "right" | "left" | "top-start" | "top-end" | "bottom-start" | "bottom-end" | "right-start" | "right-end" | "left-start" | "left-end"` will instruct the Overlay where to place itself in relationship to the trigger element.
+
+### receivesFocus
+
+Some Overlays will always be passed focus (e.g. modal or page Overlays). When this is not true, the `receivesFocus` option will inform the API whether to attempt to pass focus into the Overlay once it is open. `'true'` will pass focus, `'false'` will not (when possible), and `"auto"` (the default), will make a decision based on the `type` of the Overlay.
+
+### trigger
+
+The `trigger` option accepts an `HTMLElement` or a `VirtualTrigger` from which to position the Overlay.
```

Comment:
Several of these options reference a trigger. Should we specify here the side-effects of not providing a trigger, with an example of why you'd use or not use one?